x/upgrade/keeper/keeper_test.go

```diff
@@ -194,20 +194,22 @@ func (s *KeeperTestSuite) TestSetUpgradedClient() {
 
 }
 
-// Test that the protocol version successfully increments after an
+// Test that the app version successfully increments after an
 // upgrade and is successfully set on BaseApp's appVersion.
-func (s *KeeperTestSuite) TestIncrementProtocolVersion() {
-	oldProtocolVersion := s.app.BaseApp.AppVersion()
+func (s *KeeperTestSuite) TestIncrementAppVersion() {
+	oldAppVersion, err := s.app.BaseApp.GetAppVersion(s.ctx)
+	s.Require().NoError(err)
```

Comment:
Should we add a test here for what happens if there is an error. (Which will be the case for Osmosis)

or does Osmosis directly set this via some other mechanism

Reply:
Upon further investigation, this test is impossible because this is an integration suite level test. That is, the app version gets initialized to 0 in either [`InitChain`](https://github.com/osmosis-labs/cosmos-sdk/blob/bd68e6b0e611b8f9c5c68676b9ed8f7ef32866ab/baseapp/abci.go#L48) or in [Baseapp.init(...)](https://github.com/osmosis-labs/cosmos-sdk/blob/bd68e6b0e611b8f9c5c68676b9ed8f7ef32866ab/baseapp/baseapp.go#L321) that is called during `LoadVersion(...)`. As a result, an error here should never happen